crates/ruff_python_formatter/src/comments/visitor.rs

```diff
@@ -658,6 +658,17 @@ impl<'a> CommentPlacement<'a> {
             comment: comment.into(),
         }
     }
+
+    /// Chains the placement with the given function.
+    ///
+    /// Returns `self` when the placement is non-[`CommentPlacement::Default`]. Otherwise, calls the
+    /// function with the comment and returns the result.
+    pub(super) fn then_with<F: FnOnce(DecoratedComment<'a>) -> Self>(self, f: F) -> Self {
```

Comment:
Nit: I think I would prefer `or_else` to match the `Option` and `Result` naming (which is what we used to have before)

Reply:
Interesting, this to me is more similar to the ordering semantics so following their naming felt more natural to me, I will change though.